cf/data/creation.py

```diff
@@ -81,22 +82,46 @@ def to_dask(array, chunks, **from_array_options):
 
     **Examples**
 
-    >>> to_dask([1, 2, 3])
+    >>> cf.data.creation.to_dask([1, 2, 3])
     dask.array<array, shape=(3,), dtype=int64, chunksize=(3,), chunktype=numpy.ndarray>
-    >>> to_dask([1, 2, 3], chunks=2)
+    >>> cf.data.creation.to_dask([1, 2, 3], chunks=2)
     dask.array<array, shape=(3,), dtype=int64, chunksize=(2,), chunktype=numpy.ndarray>
-    >>> to_dask([1, 2, 3], chunks=2, {'asarray': True})
+    >>> cf.data.creation.to_dask([1, 2, 3], chunks=2, {'asarray': True})
     dask.array<array, shape=(3,), dtype=int64, chunksize=(2,), chunktype=numpy.ndarray>
+    >>> cf.data.creation.to_dask(cf.dt(2000, 1, 1), 'auto')
+    dask.array<array, shape=(), dtype=object, chunksize=(), chunktype=numpy.ndarray>
+    >>> cf.data.creation.to_dask([cf.dt(2000, 1, 1)], 'auto')
+    dask.array<array, shape=(1,), dtype=object, chunksize=(1,), chunktype=numpy.ndarray>
 
     """
+    if not (
+        is_dask_collection(array)
+        or isinstance(
+            array, (np.ndarray, list, tuple, memoryview) + np.ScalarType
+        )
+    ):
+        if not hasattr(array, "shape"):
```

Comment:
Is there a reason you chose not to combine the two `if` statements into one here, for foreseen extensibility perhaps?

Reply:
No reason. Rolled into one test: 67113887cfd2866978b6ead41b788ec25e2c8e6e